pkg/apis/infinispan/v1/types_util.go

```diff
@@ -94,8 +94,9 @@ func (ispn *Infinispan) ApplyDefaults() {
 		if ispn.Spec.Service.Container == nil {
 			ispn.Spec.Service.Container = &InfinispanServiceContainerSpec{}
 		}
-		if ispn.Spec.Service.Container.Storage == "" {
-			ispn.Spec.Service.Container.Storage = consts.DefaultPVSize.String()
+		if ispn.Spec.Service.Container.Storage == nil {
+			def := consts.DefaultPVSize.String()
+			ispn.Spec.Service.Container.Storage = &def
```

Comment:
No need for the variable, we can just do this in a one-liner.

Reply:
Fixed